ISPN-5378 Additional Lucene Directory Wildfly integration tests

alanfx opened the pull request:
https://issues.jboss.org/browse/ISPN-5378

Added IndexReader and Spatial search tests

IndexReader tests are working
Spatial search are mostly working. The only test that is failing is the
long distance search with a projection. It is only returning one member
instead of two, as the query without a projection does.

This PR should get integrated on the master and 7.2.x branches after https://github.com/infinispan/infinispan/pull/3448 is integrated.

alanfx commented:
@Sanne and @gustavonalle here are some more tests

rhusar commented on integrationtests/as-lucene-directory/src/test/java/org/infinispan/test/integration/as/wildfly/model/Member.java:
```diff
@@ -27,7 +37,30 @@
+   Double latitude;
+
+   @Longitude
+   Double longitude;
```
I guess these 2 rather be private?

alanfx replied:
Good point, @rhusar. I added the getter and setter later

Sanne replied:
there's actually no need.
Some people prefer the strong encapsulation and "traditional Java" style, like beans. But some others preferring concise and to the point code will use public field accessors.. both are supported, so it's actually a good idea to test for both in our tests.

alanfx replied:
Ugh, thanks @Sanne! I'll revert my last commit. I guess that's what I get for listening to @rhusar! :-)

rhusar replied:
My rationale is that we have seen people going to the project integration tests to figure out how the functionality actually works and how to use it best and thus we should use the best practices. Though, the point about testing is good. But im still not on the breaking encapsulation boat. :)

galderz commented:
Marking it as preview since some tests are still failing:

```
org.infinispan.test.integration.as.wildfly (4)
  InfinispanModuleMemberRegistrationIT.testNewMemberSpatialSearchLongDistance 	
  InfinispanModuleMemberRegistrationIT.testNewMemberSpatialSearchNearCBGB 	
  InfinispanModuleMemberRegistrationIT.testNewMemberSpatialSearchNearGD 	
  InfinispanModuleMemberRegistrationIT.testReIndex 	
```

alanfx commented:
Hey @galderz, these failures will go away when https://github.com/infinispan/infinispan/pull/3448 is integrated. (Well,  testNewMemberSpatialSearchLongDistance will still fail)

gustavonalle commented:
@alanfx #3448 was merged

alanfx commented:
Thanks @gustavonalle. Rebased this PR, so it should be ready to review.

gustavonalle commented:
@alanfx  `InfinispanModuleMemberRegistrationIT.testNewMemberSpatialSearchLongDistance` is failing, have you got a fix?

alanfx commented:
@gustavonalle no, I don't. I mentioned this in the description of the PR. The test runs the same spatial query twice. With spatialSearch() it passes, but with spatialSearchWithDistance() which gets the distance projection; it fails. I can comment that part of the code out, if you like.

galderz commented:
@gustavonalle @alanfx What are we doing about this?

gustavonalle commented:
@galderz  We can integrate this, if @alanfx ignores the failing test and open a JIRA for the projection issue.

alanfx commented:
@gustavonalle, @Sanne, and @galderz PR is rebased and updated. I split the tests with and without the projection, and then added an "Ignore" annotation to the failing test. I created HSEARCH-1918 for the issue I am seeing.

alanfx commented:
@gustavonalle and @Sanne this PR has been rebased and after some changes suggested by Nicolas Helleringer in HSEARCH-1918, all tests pass! Can I get some review love?!

gustavonalle commented:
Looks good, integrating...

gustavonalle commented:
Merged, thanks @alanfx

alanfx commented:
Thanks, @gustavonalle!
